src/coreclr/jit/lclmorph.cpp

```diff
@@ -936,18 +936,51 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 var_types elementType = indir->TypeGet();
                 lclNode               = indir->gtGetOp1()->BashToLclVar(m_compiler, lclNum);
 
-                if (elementType == TYP_FLOAT)
+                switch (elementType)
                 {
-                    GenTree* indexNode = m_compiler->gtNewIconNode(offset / genTypeSize(elementType));
-                    hwiNode = m_compiler->gtNewSimdGetElementNode(elementType, lclNode, indexNode, CORINFO_TYPE_FLOAT,
-                                                                  genTypeSize(varDsc));
-                }
-                else
-                {
-                    assert(elementType == TYP_SIMD12);
-                    assert(genTypeSize(varDsc) == 16);
-                    hwiNode = m_compiler->gtNewSimdHWIntrinsicNode(elementType, lclNode, NI_Vector128_AsVector3,
-                                                                   CORINFO_TYPE_FLOAT, 16);
+                    case TYP_FLOAT:
+                    {
+                        GenTree* indexNode = m_compiler->gtNewIconNode(offset / genTypeSize(elementType));
+                        hwiNode            = m_compiler->gtNewSimdGetElementNode(elementType, lclNode, indexNode,
+                                                                      CORINFO_TYPE_FLOAT, genTypeSize(varDsc));
+                        break;
+                    }
+                    case TYP_SIMD12:
+                    {
+                        assert(genTypeSize(varDsc) == 16);
+                        hwiNode = m_compiler->gtNewSimdHWIntrinsicNode(elementType, lclNode, NI_Vector128_AsVector3,
+                                                                       CORINFO_TYPE_FLOAT, 16);
+                        break;
+                    }
+#if defined(FEATURE_SIMD) && defined(TARGET_XARCH)
+                    case TYP_SIMD16:
```

Comment:
Do we need to handle `TYP_SIMD16` for `Arm64` since it has `TYP_SIMD8` (`Vector64<T>`) fields?

Reply:
Sure, I've added it for consistency.
On a side note I've also removed the quarter handling since it had no diffs and hence no coverage. I think if we want that we should add it as a separate PR that also adds some adequate testing.

Reply:
With it removed, would we assert if some code did hit that code path?

Or is that mostly limited by the fact that our software fallbacks primarily defer to the next narrower size's implementation, which will be intrinsic on hardware  that does support the current size?

Reply:
> With it removed, would we assert if some code did hit that code path?

No, this code is tied together with the code in `SelectLocalIndirTransform`. I added some code there to compensate -- it only returns `GetElement`/`WithElement` for halves.